apps/alert_processor/lib/model/query.ex

```diff
@@ -82,6 +82,58 @@ defmodule AlertProcessor.Model.Query do
         LEFT JOIN subscriptions s ON u.id = s.user_id
         GROUP BY u.communication_mode
         """
+      },
+      %__MODULE__{
+        label: "Weekly abandonment rate",
+        query: """
+        WITH
+          periods AS (
+            SELECT \"end\" - interval '1 week' AS start, \"end\"
+            FROM generate_series(
+              date_trunc('week', (SELECT MIN(inserted_at) FROM users)) - interval '1 second',
+              date_trunc('week', NOW() AT TIME ZONE 'utc') + interval '1 week' - interval '1 second',
+              interval '1 week'
+            ) \"end\"
+          ),
+          new_users_per_period AS (
+            SELECT COUNT(DISTINCT users.id) AS "count", periods.end AS period_end
+            FROM users
+            INNER JOIN periods ON (
+              periods.start <= users.inserted_at
+              AND users.inserted_at <= periods.end
+            )
+            WHERE
+              users.email_rejection_status IS NULL
+            GROUP BY
+              periods.end
+          ),
+          new_users_without_subscriptions_per_period AS (
+            SELECT COUNT(DISTINCT users.id) AS "count", periods.end AS period_end
+            FROM users
+            INNER JOIN periods ON (
+              periods.start <= users.inserted_at
+              AND users.inserted_at <= periods.end
+            )
+            LEFT JOIN subscriptions ON subscriptions.user_id = users.id
+            WHERE
+              users.email_rejection_status IS NULL
+              AND subscriptions.id IS NULL
+            GROUP BY
+              periods.end
+          )
+          SELECT
```

Comment:
Minor indentation issue: I think this line and all the following ones should be de-indented one level (as this is the end of the `WITH` clause that introduced the extra level of indentation).

Reply:
👀